Update releasing doc about re-enabling nightly tests (#2791)
adoptium/temurin-build: RELEASING.md

     - [ci-jenkins-pipelines](https://github.com/adoptium/ci-jenkins-pipelines) (for jenkins pipelines specific issues)
     - [eclipse/openj9](https://github.com/eclipse-openj9/openj9) (for OpenJ9 issues)
5. Discuss failing tests with [Shelley Lambert](https://github.com/smlambert)

> 6. Once all AQA tests on all platforms have been signed off, then nightly tests can be re-enabled.

Reviewer: ```suggestion
6. Once all AQA tests on all platforms have been signed off, then nightly tests can be re-enabled. See the notes on checking the regeneration job worked in step 1
```

Author: added

> 7. If "good to publish", then get permission to publish the release from the Adoptium PMC members, discussion is via the Adoptium [#release](https://adoptium.slack.com/messages/CLCFNV2JG) Slack channel and create a Promotion TSC item [here](https://github.com/AdoptOpenJDK/TSC/issues/new?assignees=&labels=&template=promote-release.md&title=Promote+AdoptOpenJDK+Version+%3Cx%3E).

Reviewer: I think we should remove the bits from `and create a Promotion TSC item` since we haven't been doing that (it's all been via tagging `@pmc` in the #release slack channel, also the TSC repo at AdoptOpenJDK wouldn't be the right place for it now anyway (If we wanted to, `adoptium/adoptium` would probably be the place to do it.
```suggestion
7. If "good to publish", then get permission to publish the release from the Adoptium PMC members, discussion is via the Adoptium [#release](https://adoptium.slack.com/messages/CLCFNV2JG) Slack channel by tagging `@pmc` asking for permission.
```

Author: fixed
